pkg/cloud/cloudtestutils/cloud_test_helpers.go

```diff
@@ -399,81 +399,133 @@ func CheckListFilesCanonical(t *testing.T, info StoreInfo, canonical string) {
 	}
 
 	t.Run("List", func(t *testing.T) {
+		// NB: When using AfterKey in tests, pick keys such that they do not exist
+		// in the expected output to avoid tests flaking due to different cloud
+		// provider behaviors around AfterKey's inclusiveness.
 		for _, tc := range []struct {
-			name      string
-			uri       string
-			prefix    string
-			delimiter string
-			expected  []string
+			name     string
+			uri      string
+			prefix   string
+			opts     cloud.ListOptions
+			expected []string
 		}{
 			{
 				"root",
 				info.URI,
 				"",
-				"",
+				cloud.ListOptions{},
 				foreach(fileNames, func(s string) string { return "/" + s }),
 			},
 			{
 				"file-slash-numbers-slash",
 				info.URI,
 				"file/numbers/",
-				"",
+				cloud.ListOptions{},
 				[]string{"data1.csv", "data2.csv", "data3.csv"},
 			},
 			{
 				"root-slash",
 				info.URI,
 				"/",
-				"",
+				cloud.ListOptions{},
 				foreach(fileNames, func(s string) string { return s }),
 			},
 			{
 				"file",
 				info.URI,
 				"file",
-				"",
+				cloud.ListOptions{},
 				foreach(fileNames, func(s string) string { return strings.TrimPrefix(s, "file") }),
 			},
 			{
 				"file-slash",
 				info.URI,
 				"file/",
-				"",
+				cloud.ListOptions{},
 				foreach(fileNames, func(s string) string { return strings.TrimPrefix(s, "file/") }),
 			},
 			{
 				"slash-f",
 				info.URI,
 				"/f",
-				"",
+				cloud.ListOptions{},
 				foreach(fileNames, func(s string) string { return strings.TrimPrefix(s, "f") }),
 			},
 			{
 				"nothing",
 				info.URI,
 				"nothing",
-				"",
+				cloud.ListOptions{},
 				nil,
 			},
 			{
 				"delim-slash-file-slash",
 				info.URI,
 				"file/",
-				"/",
+				cloud.ListOptions{Delimiter: "/"},
 				[]string{"abc/", "letters/", "numbers/"},
 			},
 			{
 				"delim-data",
 				info.URI,
 				"",
-				"data",
+				cloud.ListOptions{Delimiter: "data"},
 				[]string{"/file/abc/A.csv", "/file/abc/B.csv", "/file/abc/C.csv", "/file/letters/data", "/file/numbers/data"},
 			},
+			{
+				"afterkey-no-prefix",
+				info.URI,
+				"",
+				cloud.ListOptions{AfterKey: "file/letters/dataB"},
+				[]string{"/file/letters/dataB.csv", "/file/letters/dataC.csv", "/file/numbers/data1.csv", "/file/numbers/data2.csv", "/file/numbers/data3.csv"},
+			},
+			{
+				"afterkey-with-prefix",
+				info.URI,
+				"file/letters/",
+				cloud.ListOptions{AfterKey: "file/letters/dataB"},
```

Comment:
def no need to change, but these cases would be slightly easier to read if you used the `letters` files throughout, aside from the delimiter cases below.